c/tskit/trees.c

```diff
@@ -4613,6 +4621,7 @@ tsk_tree_free(tsk_tree_t *self)
     tsk_safe_free(self->next_sample);
     tsk_safe_free(self->num_children);
     tsk_safe_free(self->edge);
+    tsk_safe_free(self->tree_pos);
```

Comment:
This is wrong , and I'm surprised valgrind doesn't catch it. The golden rule is, you only ``free`` a pointer than you have previously ``malloc``ed. The ``tree_pos`` here is an embedded struct, so it's not  a pointer that can be freed.

You should be calling ``tsk_tree_position_free(&self->tree_pos);`` instead.